code/core/src/common/js-package-manager/PNPMProxy.ts

```diff
@@ -159,13 +160,11 @@ export class PNPMProxy extends JsPackageManager {
       }
     }
 
-    const packageJsonPath = await findUpSync(
-      (dir) => {
-        const possiblePath = join(dir, 'node_modules', packageName, 'package.json');
-        return existsSync(possiblePath) ? possiblePath : undefined;
-      },
-      { cwd: basePath }
-    );
+    const dirs = walk.up(basePath ?? '.');
```

Comment:
same as in
https://github.com/storybookjs/storybook/pull/31338/files#r2075527459